core/services/flux_monitor_test.go

```diff
@@ -0,0 +1,264 @@
+
+	cltest.CallbackOrTimeout(t, "deviation checker started", func() {
+		<-started
+	})
+	checkerFactory.AssertExpectations(t)
+	dc.AssertExpectations(t)
+
+	// 2. Remove Job
+	removed := make(chan struct{})
+	dc.On("Stop").Return().Run(func(mock.Arguments) {
+		removed <- struct{}{}
+	})
+	fm.RemoveJob(job.ID)
+	cltest.CallbackOrTimeout(t, "deviation checker stopped", func() {
+		<-removed
+	})
+	dc.AssertExpectations(t)
+}
+
+func TestConcreteFluxMonitor_AddJobError(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	job := cltest.NewJobWithFluxMonitorInitiator()
+	runManager := new(mocks.RunManager)
+	dc := new(mocks.DeviationChecker)
+	dc.On("Initialize", mock.Anything).Return(errors.New("deliberate test error"))
+	checkerFactory := new(mocks.DeviationCheckerFactory)
+	checkerFactory.On("New", mock.Anything, job.Initiators[0], runManager).Return(dc, nil)
+	fm := services.NewFluxMonitor(store, runManager)
+	services.ExportedSetCheckerFactory(fm, checkerFactory)
+	require.NoError(t, fm.Connect(nil))
+	defer fm.Disconnect()
+
+	require.Error(t, fm.AddJob(job))
+	checkerFactory.AssertExpectations(t)
+	dc.AssertExpectations(t)
+}
+
+func TestConcreteFluxMonitor_AddJobDisconnected(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	job := cltest.NewJobWithFluxMonitorInitiator()
+	runManager := new(mocks.RunManager)
+	checkerFactory := new(mocks.DeviationCheckerFactory)
+	fm := services.NewFluxMonitor(store, runManager)
+	services.ExportedSetCheckerFactory(fm, checkerFactory)
+
+	require.Error(t, fm.AddJob(job))
+}
+
+func TestConcreteFluxMonitor_ConnectStartsExistingJobs(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	job := cltest.NewJobWithFluxMonitorInitiator()
+	require.NoError(t, store.CreateJob(&job))
+
+	job, err := store.FindJob(job.ID) // Update job from db to get matching coerced time values (UpdatedAt)
+	require.NoError(t, err)
+
+	runManager := new(mocks.RunManager)
+	started := make(chan struct{})
+
+	dc := new(mocks.DeviationChecker)
+	dc.On("Initialize", mock.Anything).Return(nil)
+	dc.On("Start").Return().Run(func(mock.Arguments) {
+		started <- struct{}{}
+	})
+
+	checkerFactory := new(mocks.DeviationCheckerFactory)
+	checkerFactory.On("New", mock.Anything, job.Initiators[0], runManager).Return(dc, nil)
+	fm := services.NewFluxMonitor(store, runManager)
+	services.ExportedSetCheckerFactory(fm, checkerFactory)
+	require.NoError(t, fm.Connect(nil))
+	cltest.CallbackOrTimeout(t, "deviation checker started", func() {
+		<-started
+	})
+	checkerFactory.AssertExpectations(t)
+	dc.AssertExpectations(t)
+}
+
+func TestPollingDeviationChecker_PollHappy(t *testing.T) {
+	fetcher := new(mocks.Fetcher)
+	fetcher.On("Fetch").Return(102.0, nil)
+
+	job := cltest.NewJobWithFluxMonitorInitiator()
+	initr := job.Initiators[0]
+	initr.ID = 1
+
+	rm := new(mocks.RunManager)
+	run := cltest.NewJobRun(job)
+	data := models.JSON{Result: gjson.Parse(`{"result":"102"}`)}
```

Comment:
why not use our models.ParseJSON() here?  This way we cant remove the dependency of gjson